hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoDeleteAllowanceHandler.java

```diff
@@ -61,9 +62,10 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         final var txn = context.body();
         pureChecks(txn);
         final var op = txn.cryptoDeleteAllowanceOrThrow();
-        // Every owner whose allowances are being removed should sign, if the owner is not payer
+        // Every owner whose allowances are being removed should sign (or the payer, if there is no owner)
+        final var payerId = txn.transactionID().accountID();
         for (final var allowance : op.nftAllowancesOrElse(emptyList())) {
-            context.requireKeyOrThrow(allowance.ownerOrElse(AccountID.DEFAULT), INVALID_ALLOWANCE_OWNER_ID);
+            context.requireKeyOrThrow(allowance.ownerOrElse(payerId), INVALID_ALLOWANCE_OWNER_ID);
```

Comment:
Payer would have already signed in workflow right ? Why to sign again ? It should just be  

```
for (final var allowance : op.nftAllowancesOrElse(emptyList())) {
            if(allowance.hasOwner()){
                context.requireKeyOrThrow(allowance.ownerOrThrow(), INVALID_ALLOWANCE_OWNER_ID); 
            }
        }
``` ?

Reply:
The payer would have signed already, you're right

Reply:
Fixed